### DISCUSSION TITLE
Welcome to Ubuntu-on-rpi-fixes Discussions!

### DISCUSSION THREAD

**Discussion by Itai-Nelken** [Itai-Nelken/Ubuntu-on-rpi-fixes#1] [General]
## 👋 Welcome!
  
I opened this so you can ask questions without opening an issue.
please don't use this for general chat, but only support for my scripts.
this is <b>not</b> a replacement for issues by any means.

We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

<b>Thank you for reading!</b>

**Comment by searider74** [comment]
Hello Itai,

I have used your procedures to install box86 and zoom on my rpi4.  However, I have discovered a few things that need correction on my rpi4.  I am running Ubuntu 21.10 on my rpi4 with 4 GB of memory.

Regarding your procedures.  First, I noticed that once I performed the following commands, I no longer get sound from bluetooth or to my hdmi speakers.

      sudo dpkg --add-architecture armhf
      sudo apt update

When trying to connect to my bluetooth speaker (which was working prior to this procedure), I now get a "protocol failure" message and bluetooth fails to connect.  The bluetooth daemons are running with errors based on:

      systemctl status bluetooth

But no connection is made.  I also noticed that pulseaudio no longer gets started on reboots.  I can start it manually, but still, no sound from any device.  I suspect that the system is now getting confused between the appropriate arm64 and armhf libraries when it comes to sound.  Have you experienced this issue?  Any ideas on how to get my sound back while still being able to run zoom and my normal things?

Second, regarding the use of the tilda "~" character in your paths.  This did not work for me.  It is bad practice to use the tilda in paths because the order in which bash does path, redirection, and substitutions in shells and subshells can sometimes lead to strange and unpredictable results.  It is better to use $HOME or full pathnames in your "startzoom.sh" script and the "zoom.desktop" shortcut.  After making such changes, I was able to start zoom via box86.  The video portion seems to work fine, but no sound.

Thanks for the work you have done on this project.

Searider74

**Reply by Itai-Nelken** [reply]
Hi,
I wrote this tutorial a long time ago (Ubuntu 20.04 times), and I updated it a bit when 20.10 and 21.04 were released. I didn't update it since so some stuff might not work.
About the bluetooth problem you have - you are probably right about the armhf and arm64 packages conflicting, I think it might be `pulseaudio`. this problem has never happened to me, but bluetooth was never very stable on ubuntu on rpi for me.
You are completely right about using `$HOME` instead of `~`, I'll change this when I have time.
About Zoom - use `box64` and x64 zoom, it works much better and audio input works!

I don't have time for fixing and testing everything, life is in the way... but once I have time I'll install a fresh 21.04 (or whatever is the newset version then) and test every tutorial and fix any stuff that doesn't work.
If you can help I'll appreciate it a lot.

Thanks for reaching out and reminding me that I have to fix stuff.

**Reply by searider74** [reply]
I fully understand about how long ago you wrote the tutorial and the lack of time due to, well..."life"!

Updates since my initial post:  I am still not able to get bluetooth working again.  However, I am now getting sound out of my hdmi monitor speakers.  And this was vastly improved for clarity when configuring "load-module module-udev-detect tsched=0" in the /etc/pulse/default.pa file, then rebooting.  But it seems, either zoom or box86 is not configuring or finding my microphone properly, no matter if via a usb connected miicrophone or the webcam mic.  From your response regarding audio input working above, it seems you have also run into this issue.  Am I correct?
I am continuing to try to track this down or find a working alternative.  However, with Box64 and x86 zoom, non of this may no longer be necessary.

Something else you should know about my system.  I run Ubuntu Mate Desktop.  Bluetooth was pretty stable on this setup up until Ubuntu Release 21.04.  It runs quite well 20.10 and 20.04, but somewhere along the line, a failure was introduced in 21.04 that I suspect is with pulseaudio.  Pulseautio in 21.04 Ubuntu Mate no longer recognizes the hdmi monitor speakers, but it does recognize and work with bluetooth speakers.  It also works with the built-in audio.  As a result, I have stayed on 20.10 and am looking forward to 21.10 coming out in a few days.  Hopefully, they have fixed this issue with pulseaudio.

I will also install box64 and give that a try.  Do you recommend the ubuntu 64 bit download for x64 zoom or just to download the tar file for x64 zoom?

Finally, I would be happy to help test things if I can.  I am not a programmer but I'm not a newbie around Linux either.  So, if there is something I can help you with, let me know.  I will tell you then if it is beyond me or not.

Searider74

**Reply by Itai-Nelken** [reply]
Use the x64 tar for zoom.
